tests/test_meshtastic_utils_async.py

```
@@ -558,3 +592,245 @@ def mock_create_task(coro_arg):
+        timeout_seconds=0.1,
+        poll_seconds=0.01,
+    )
+
+    assert result == "ok"
+
+
+def test_wait_for_future_result_with_shutdown_aborts_when_shutting_down():
+    future = MagicMock()
+
+    with patch.object(mu, "shutting_down", True):
+        with pytest.raises(TimeoutError, match="Shutdown in progress"):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)